src/fluree/db/full_text.clj

```diff
@@ -158,25 +158,49 @@
   (doto wrtr .deleteAll .commit)
   (block-registry/reset block-registry))
 
+(defn parse-domain
+  [search]
+  (-> search
+      (str/split #"^fullText:")
+      second))
+
+(defn predicate-domain?
+  [domain]
+  (str/includes? domain "/"))
+
+(defn build-predicate-query
+  [db pred param]
+  (let [pid (dbproto/-p-prop db :id pred)]
+    {pid param}))
+
+(defn build-collection-query
+  [db coll param]
+  (let [cid    (dbproto/-c-prop db :id coll)
+        params (->> (full-text-predicates db cid)
+                    (map (fn [pid]
+                           {pid param}))
+                    (into #{}))]
+    [{:_collection cid} params]))
+
+(defn build-query
+  [db domain param]
+  (if (predicate-domain? domain)
+    (build-predicate-query db domain param)
+    (build-collection-query db domain param)))
+
+(defn wildcard?
```

Comment:
Is there a known way to escape this, if you really wanted to find the character '*' or '?'? Not urgent at the moment, but if so it might be useful to incorporate it into a test for at least a minimal documentation that it exists.

Reply:
We could escape it ourselves in this function. The code in this patch only looks for a '*' or '?' to decide if it's a wildcard and then calls the wildcard api. We could introduce an escape character like '\' to indicate we want to search for a literal '*' or '?' and strip out the slashes and use the normal `search` api. I can make that change, but there might be some edge cases.

Reply:
I don't think it is worth doing... and the wildcard should also capture any values that just had an asterisk, so I think the current behavior is fine. If someone has a use case in the future that presents an issue we can look at solutions then.

Reply:
I just thought of an edge case as I started writing the code. If someone wants to search for both a wildcard in one part of the query and a literal '\*' in another part, then I'm not sure how we can make it work because of the two different query apis we'd need to use. 

I think I'll add another ticket for this specific issue since it isn't urgent at the moment and I think we'll have to work through all the edge cases. I'll also add another to revisit our choice of full text library since Clucie is so outdated.

Reply:
I think looking at additional libraries is good. It does appear we use a fairly small surface of the Lucene API and I'm guessing it might make most sense just to use java interop directly instead of dealing with other layers. But a ticket to look at this sounds great.